File: app/src/main/java/zachwhaley/atwork/AtWorkLocationService.java
package zachwhaley.atwork;

import android.app.IntentService;
import android.content.Intent;
import android.os.Bundle;

public class AtWorkLocationService extends IntentService {

    public AtWorkLocationService(String name) {
        super(name);
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        AtWork.log("AtWorkLocationService.onHandleIntent");
        // TODO: Figure out which one of these values is the location object.
        Bundle bundle = intent.getExtras();
        for (String key : bundle.keySet()) {
            AtWork.log(bundle.get(key).toString());
        }
        AtWork.sendNotification(this, "Location found!");
        AtWorkLocationReceiver.completeWakefulIntent(intent);
    }
}

File: app/src/main/java/zachwhaley/atwork/AtWorkLocationService.kt
package zachwhaley.atwork

import android.app.IntentService
import android.content.Intent
import android.support.v4.content.WakefulBroadcastReceiver

class AtWorkLocationService : IntentService("AtWorkLocationService") {

    override fun onHandleIntent(intent: Intent?) {
        AtWork.log("AtWorkLocationService.onHandleIntent")
        // TODO: Figure out which one of these values is the location object.
        intent?.let {
            for (key in it.extras.keySet()) {
                AtWork.log(it.extras.get(key)?.toString())
            }
        }
        WakefulBroadcastReceiver.completeWakefulIntent(intent)
    }
}
